packages/caliper-fabric/test/connector-versions/peer-gateway/PeerGateway.js

```diff
@@ -0,0 +1,498 @@
+    });
+
+    it('should return a context when preparing for use by a caliper worker', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        const context = await peerGateway.getContext();
+        context.should.be.instanceOf(FabricConnectorContext);
+    });
+
+    it('should return the same context when requested multiple times', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        const context = await peerGateway.getContext();
+        const context2 = await peerGateway.getContext();
+        context2.should.equal(context);
+    });
+
+    it('should pass the default timeout in the gateway options', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        const now = new Date();
+        const clock = sinon.useFakeTimers(now.getTime());
+        await peerGateway.getContext();
+        Gateway.connectArgs[0].evaluateOptions().should.be.deep.equal({deadline: now.getTime() + 60000});
+        clock.restore();
+    });
+
+    it('should pass the timeout from invokeOrQuery in the gateway options', async () => {
+        ConfigUtil.set(ConfigUtil.keys.Fabric.Timeout.InvokeOrQuery, 99);
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        const now = new Date();
+        const clock = sinon.useFakeTimers(now.getTime());
+        await peerGateway.getContext();
+        Gateway.connectArgs[0].evaluateOptions().should.be.deep.equal({deadline: now.getTime() + 99000});
+        clock.restore();
+        ConfigUtil.set(ConfigUtil.keys.Fabric.Timeout.InvokeOrQuery, null);
+    });
+
+    it('should create one Client for each organization', async () => {
+        const connectorConfiguration = await new ConnectorConfigurationFactory().create(path.resolve(__dirname, configWith2Orgs1AdminInWalletNotMutual), walletFacadeFactory);
+        const peerGateway = new PeerGateway(connectorConfiguration, 1, 'fabric');
+        await peerGateway.getContext();
+        GrpcClient.constructed.should.equal(2);
+        GrpcClient.options[0]['grpc.hostnameOverride'].should.deep.equal('peer0.org1.example.com');
```

Comment:
I would remove these option test lines and create a new test that states
`should create a client with the specified tls options` and check the tls options in that test